### DISCUSSION TITLE
Que las states de una SM queden ordenadas alfabéticamente

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#356] [3. Ideas & Feature Requests]
Hola

A veces sucede que en el comienzo de una SM, se crean muy pocos estados y enumerándolas como 1,2,3,4, respectivamente. Sin embargo cuando se realizan modificaciones y se agregan nuevos estados o editar los ya existentes, a veces queda como 1,9,2,4,3,5,6, dejando la lista desordenada y con una dificultad en la hora de revisar las rutinas de éstas.

**Dejo un ejemplo cuando son pocos estados y hayan tenido alguna edición.**
![image](https://user-images.githubusercontent.com/61419471/127223854-51331831-16ef-4bd7-9df4-641f4565a70d.png)

**Dejo un ejemplo cuando son muchos estados y muchas modificaciones.**
![image](https://user-images.githubusercontent.com/61419471/127224491-f608ab0e-3b3b-486e-a895-29df6bdbcbe5.png)

Son 27 estados aprox y se están utilizando alrededor de 15.

Existe la posibilidad que se pueda realizar ese ordenamiento?

Gracias =>

**Comment by acedward** [comment]
En este mismo momento no tenemos forma de hacerlo a través de la UI.
Pero suena absolutamente razonable que deberían tener un orden - similar al del grafo de la máquina de estado, u en forma alfabética. Tenemos un proyecto interno para mejorar esta visualización, pero aún está en backlog.

Deja esto abierto - por si alguién más tiene alguna idea para un workaround.

**Reply by guillevalin** [reply]
@acedward estaba explorando hacerlo desde el front-end y se veia relativamente sencillo aunque la forma correcta de hacerlo probablemente sea que el resultado sea ordenable por cada una de las columnas en todas las vistas de este estilo en el admin.

**Reply by acedward** [reply]
@guillevalin es que no necesariamente es alfabético - porque el estado new debería estar primero? - creo que debería ser el orden BFS
